lib/secretsscanner/reporter/report.go

```diff
@@ -0,0 +1,211 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package reporter
+
+import (
+	"context"
+	"errors"
+	"io"
+	"log/slog"
+
+	"github.com/gravitational/trace"
+
+	accessgraphsecretsv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/accessgraph/v1"
+	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+	dtassert "github.com/gravitational/teleport/lib/devicetrust/assert"
+	secretsscannerclient "github.com/gravitational/teleport/lib/secretsscanner/client"
+)
+
+const (
+	defaultBatchSize = 100
+)
+
+// AssertCeremonyBuilderFunc is a function that builds the device authentication ceremony.
+type AssertCeremonyBuilderFunc func() (*dtassert.Ceremony, error)
+
+// Config specifies the configuration for the reporter.
+type Config struct {
+	// ProxyAddress is the address of the proxy server to send the report to.
+	ProxyAddress string
+	// InsecureSkipVerify specifies whether to trust the certificates from the Proxy Server during registration without verification.
+	InsecureSkipVerify bool
+	// Log is the logger.
+	Log *slog.Logger
+	// BatchSize is the number of secrets to send in a single batch. Defaults to [defaultBatchSize] if not set.
+	BatchSize int
+	// AssertCeremonyBuilder is the device authentication ceremony builder.
+	// If not set, the default device authentication ceremony will be used.
+	AssertCeremonyBuilder AssertCeremonyBuilderFunc
+}
+
+// Reporter reports secrets to the Teleport Proxy.
+type Reporter struct {
+	proxyAddr             string
+	insecureSkipVerify    bool
+	log                   *slog.Logger
+	batchSize             int
+	assertCeremonyBuilder AssertCeremonyBuilderFunc
+}
+
+// New creates a new reporter instance.
+func New(cfg Config) (*Reporter, error) {
+	if cfg.ProxyAddress == "" {
+		return nil, trace.BadParameter("missing proxy address")
+	}
+	if cfg.Log == nil {
+		cfg.Log = slog.Default()
+	}
+	if cfg.BatchSize == 0 {
+		cfg.BatchSize = defaultBatchSize
```

Comment:
Suggestion: Since it's only relevant there:

```suggestion
		const defaultBatchSize = 100
		cfg.BatchSize = defaultBatchSize
```

and drop the package-level const.